deps.windows/70-ducible.ps1

```diff
@@ -0,0 +1,109 @@
+param(
+    [string] $Name = 'ducible',
+    [string] $Version = 'v1.2.2',
+    [string] $Uri = 'https://github.com/jasonwhite/ducible.git',
+    [string] $Hash = 'b7810415529f801d98203cba00db969a6ef88a14',
+    [array] $Patches = @(
+        @{
+            PatchFile = "${PSScriptRoot}/patches/ducible/0001-Remove-8.1-target-platform.patch"
+            HashSum = 'f45851456fd93e4491e42f108ff19e60c0de3df35f1141208c95623edae14517'
+        }
+        @{
+            PatchFile = "${PSScriptRoot}/patches/ducible/0002-Use-py-instead-of-python.patch"
+            HashSum = '5922aa7639b2a364c0f90541dc15c05dc804ace9178a40a661d360bb961584c4'
+        }
+    )
+)
+
+function Setup {
+    Setup-Dependency -Uri $Uri -Hash $Hash -DestinationPath $Path
+}
+
+function Patch {
+    Log-Information "Patch (${Target})"
+    Set-Location $Path
+
+    $Patches | ForEach-Object {
+        $Params = $_
+        Safe-Patch @Params
+    }
+}
+
+function Clean {
+    Set-Location $Path
+
+    if ( Test-Path "build_${Target}" ) {
+        Log-Information "Clean build directory (${Target})"
+        Remove-Item -Path "build_${Target}" -Recurse -Force
+    }
+}
+
+function Build {
+    Log-Information "Build (${Target})"
+    Set-Location $Path
+
+    if ( $Configuration -match "Debug" ) {
+        $DucibleConfig = 'Debug'
+    } else {
+        $DucibleConfig = 'Release'
+    }
+
+    if ( $Target -match "x64" ) {
+        $DucibleTarget = 'x64'
+    } else {
+        $DucibleTarget = 'Win32'
+    }
+
+    $Options = @(
+        '/t:Build'
+        "/p:OutDir=.\..\..\..\build_${Target}\"
+        "/p:Configuration=${DucibleConfig}"
+        "/p:Platform=${DucibleTarget}"
+        '/p:PlatformToolset=v143'
+    )
+    # Find MSBuild command
+    $MSBUILD = &"${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -latest -prerelease -products * -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe
```

Comment:
Use `Invoke-Devshell` instead: 

https://github.com/obsproject/obs-deps/blob/9d4d2fdc5b1f840d74a483e3c61aa15163079366/deps.windows/40-detours.ps1#L40

With that you should just be able to pass `msbuild -m <SOLUTION> @Options` as the `BuildCommand`.

Reply:
How did I miss that?

Reply:
Unfortunately, running the build command from within the devshell seems to mess up the environment so that external tools are no longer found when running `msbuild`. Also happens when running the dev shell manually. Confusingly, `py` and `python` still work when inside the shell, but not inside `msbuild` anymore?!

Reply:
So it doesn't even work when run in a normal dev shell either?

Reply:
If I launch the dev shell via the powershell script included with VS (`Launch-VsDevShell.ps1`) it does work. However, launching the dev shell in the same way the helper script in obs-deps does does not.

Reply:
So here's how to make it work:

* Make it depend on 60-python
* Add the python from pyenv to the path
* Run `Invoke-DevShell`
  * I used `Invoke-DevShell -BasePath $(Get-Location | Convert-Path) -BuildPath . -BuildCommand $BuildCommand -Target x64` with `msbuild -m vs\vs2015\ducible.sln /t:Build /p:Configuration=Debug /p:Platform=x64 /p:PlatformToolset=v143` as the build command
* Profit